### PR TITLE
Annotate std/file.d to please dlang/dmd#12520

### PR DESCRIPTION
**Pull request by nordlow**
Part of #8076

The https://github.com/dlang/phobos/pull/8107 and this PR are the two remaining Phobos blockers for dlang/dmd#12520.

### CODE REVIEW

**Comment by nordlow** [comment]
The error

https://app.circleci.com/pipelines/github/dlang/phobos/4283/workflows/2afbc2b3-25c7-45de-ac15-3651b9e6de90/jobs/13832

doesn't trigger locally for me when doing

`make -f posix.mak unittests `

locally in phobos root.

File: std/file.d [hunk 1]
```diff
@@ -4616,7 +4616,7 @@ enum SpanMode
     root.buildPath("animals", "dog").mkdir;
     root.buildPath("plants").mkdir;
 
-    alias removeRoot = (e) => e.relativePath(root);
+    alias removeRoot = (return scope e) @trusted => e.relativePath(root);
```
Comment (nordlow) [inline std/file.d:4619]: I need help trying to figure out the errors here. Starting with this one failing as 

```
std/file.d(4619): Error: returning `relativePath(e.name(), delegate string() pure nothrow @nogc @safe => root)` escapes a reference to parameter `e`
std/file.d(4619):        perhaps annotate the parameter with `return`
std/algorithm/iteration.d(524):        instantiated from here: `MapResult!(__lambda2, DirIterator)`
std/file.d(4621):        instantiated from here: `map!(DirIterator)`
```

File: std/file.d [hunk 2]
```diff
@@ -4616,7 +4616,7 @@ enum SpanMode
     root.buildPath("animals", "dog").mkdir;
     root.buildPath("plants").mkdir;
 
-    alias removeRoot = (e) => e.relativePath(root);
+    alias removeRoot = (return scope e) => e.relativePath(root);
```
Comment (nordlow) [inline std/file.d:4619]: Without this qualification `e` as `return scope` building with https://github.com/dlang/dmd/pull/12520 fails as

```
std/file.d(4619): Error: returning `relativePath(e.name(), delegate string() pure nothrow @nogc @safe => root)` escapes a reference to parameter `e`
std/file.d(4619):        perhaps annotate the parameter with `return`
std/algorithm/iteration.d(524):        instantiated from here: `MapResult!(__lambda2, DirIterator)`
std/file.d(4621):        instantiated from here: `map!(DirIterator)`
```

. With the qualifier dmd master fails the same way.

**Comment by nordlow** [comment]
Compiles locally now using dlang/dmd#12520. Yay! Removed WIP in title. Ready for review and auto-merge now.

**Comment by nordlow** [comment]
Does anybody know why Ubuntu 16.04 x86 and macOS 10.15 x64 fails? Restart?

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
I restarted the failing tests. Let's see.

**Pull request merged**